disk/disk_linux.go

```diff
@@ -474,7 +474,11 @@ func IOCountersWithContext(ctx context.Context, names ...string) (map[string]IOC
 		}
 		d.Name = name
 
-		d.SerialNumber, _ = SerialNumberWithContext(ctx, name)
+		// Names passed in can be full paths (/dev/sda) or just device names (sda).
+		// Since `name`` here is already a basename, re-add a hardcoded /dev path.
+		// This is not ideal, but we may break the API by changing how SerialNumberWithContext
+		// works.
+		d.SerialNumber, _ = SerialNumberWithContext(ctx, "/dev/"+name)
```

Comment:
Great Catch! But how about to use `common.HostDevWithContext` instead of hardcoded `/dev/` ?

```suggestion
		d.SerialNumber, _ = SerialNumberWithContext(ctx, common.HostDevWithContext(ctx, name))
```

Reply:
Ahh, wonderful, I missed that little helper!